index.js

```diff
@@ -76,72 +67,77 @@ module.exports = class ServeDrive extends ReadyResource {
     const { pathname, searchParams } = new URL(req.url, 'http://localhost')
 
     const id = searchParams.get('drive') // String or null
-    const drive = this.drives.get(id)
-
-    if (!drive) {
-      res.writeHead(404).end('DRIVE_NOT_FOUND')
-      return
-    }
-
-    const version = searchParams.get('checkout')
-    const snapshot = version ? drive.checkout(version) : drive
-    const filename = decodeURI(pathname)
+    const drive = await this.getDrive(id)
 
-    if (this._onfilter && !this._onfilter(id, filename)) {
-      res.writeHead(404).end('ENOENT')
-      return
-    }
-
-    let entry
     try {
-      entry = await snapshot.entry(filename)
-    } catch (e) {
-      const msg = e.code || e.message
+      if (!drive) {
+        res.writeHead(404).end('DRIVE_NOT_FOUND')
+        return
+      }
 
-      if (e.code === 'SNAPSHOT_NOT_AVAILABLE') res.writeHead(404)
-      else res.writeHead(500)
+      const version = searchParams.get('checkout')
+      const snapshot = version ? drive.checkout(version) : drive
+      if (version) res.on('close', () => snapshot.close())
```

Comment:
it used be req in node, but close on res might work now (been a while since i looked)

Reply:
Will change to req to be sure, as it's the first time I'm using this pattern.

If anyone is interested, source is here: https://nodejs.org/api/http.html#httprequesturl-options-callback . (res.close is not called in all cases)